src/components/UserSelector.tsx

```diff
@@ -1,33 +1,93 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+
+type Props = {
+  selectedUser:User | null,
+  users:User[],
+  setUser: (value: User)=> void,
+  setPosts:(value: Post[])=> void,
+  getDataFromApi: (value1: string, value2:(value: Post[])=> void) => void,
+  setDetailsSeen: (value: boolean)=> void,
+  setIsLoading: (value: boolean)=> void,
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser, users, setUser,
+  setPosts, getDataFromApi, setDetailsSeen, setIsLoading,
+}) => {
+  const [dropdownValue, setDropdownValue] = useState('Choose a user');
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+  const getUserPosts = async (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, user: User,
+  ) => {
+    event.preventDefault();
+    setIsDropdownVisible(false);
+
+    if (user.name === dropdownValue) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    setDetailsSeen(false);
+
+    await getDataFromApi('/posts', setPosts);
+    setDropdownValue(user.name);
+    setUser(user);
+  };
 
-export const UserSelector: React.FC = () => {
   return (
     <div
       data-cy="UserSelector"
-      className="dropdown is-active"
+      className={`dropdown ${isDropdownVisible && 'is-active'}`}
     >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => {
+            setIsDropdownVisible(!isDropdownVisible);
+          }}
+          onBlur={() => {
+            setTimeout(() => {
+              setIsDropdownVisible(false);
+            }, 500);
```

Comment:
Move it to a function and clear the timeout on the component unmount https://bobbyhadz.com/blog/react-cleartimeout